File: Polymorphism/Java/Calculator.java
package Polymorphism.Java;

// Calculator.java
public abstract class Calculator {
    protected int result;

    // Method to calculate the result
    public abstract void calculate(int value);

    // Method to get the current result
    public int getResult() {
        return result;
    }

    // Method to reset the result to 0
    public void reset() {
        result = 0;
    }
}

File: Polymorphism/Javascript/calculator.js
// calculator.js
class Calculator {
    constructor() {
        this._result = 0;
    }

    calculate(value) {
        throw new Error("Method 'calculate()' must be implemented.");
    }

    getResult() {
        return this._result;
    }

    reset() {
        this._result = 0;
    }
}

module.exports = Calculator;
